src/services/user.py

```
@@ -0,0 +1,784 @@
+            phone=body.phone,
+            student_id=body.student_id,
+            role=get_user_role_level("newcomer"),
+            major_id=body.major_id,
+            profile_picture=body.profile_picture,
+            profile_picture_is_url=body.profile_picture_is_url,
+        )
+
+        try:
+            user = self.user_repository.create(user)
+        except IntegrityError:
+            raise HTTPException(status_code=409, detail="unique field already exists")
```

Ruff (0.14.6)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)